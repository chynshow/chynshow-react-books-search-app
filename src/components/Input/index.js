import React from 'react';
import PropTypes from 'prop-types';

const Input = ({
  name,
  placeholder,
  type,
  onChange,
  className,
  value,
  label,
  ...rest
}) => (
  <label>
    {label}
    <input
      className={`c-input ${className ? className : ''}`}
      name={name}
      type={type}
      placeholder={placeholder}
      onChange={onChange}
      value={value}
      {...rest}
    />
  </label>
);

Input.propTypes = {
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  type: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  label: PropTypes.string,
};

Input.defaultProps = {
  type: 'text',
};

export default React.memo(Input);
