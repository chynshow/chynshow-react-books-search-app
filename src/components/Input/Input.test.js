import React from 'react';
import Input from './';
import { shallow } from 'enzyme';

const setup = (props) => shallow(<Input {...props} />);

describe('render Input component', () => {
  it('should renders Input component with props', () => {
    const c = setup({
      name: 'Name',
      placeholder: 'Placeholder',
      type: 'Type',
      onChange: () => {},
      label: 'Label',
    });
    expect(c).toMatchSnapshot();
  });
  it('should renders Input component without props', () => {
    const c = setup();
    expect(c).toMatchSnapshot();
  });
});
