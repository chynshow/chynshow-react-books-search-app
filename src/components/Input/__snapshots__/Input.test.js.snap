// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Input component should renders Input component with props 1`] = `
<label>
  Label
  <input
    className="c-input "
    name="Name"
    onChange={[Function]}
    placeholder="Placeholder"
    type="Type"
  />
</label>
`;

exports[`render Input component should renders Input component without props 1`] = `
<label>
  <input
    className="c-input "
    type="text"
  />
</label>
`;
